### PR TITLE
 Remove deprecated ProjectLayout and DefaultTask methods

### PR DESCRIPTION
**Pull request by lptr**

### CODE REVIEW

**Comment by 3flex** [comment]
Will `ProjectLayout.configurableFiles(Object...)` stay? It's also deprecated.

**Comment by lptr** [comment]
We decided to keep these around for 7.0 as anything compiled using a Gradle version earlier than 5.0 would break if we removed the workarounds.

**Comment by lptr** [comment]
`PrecompiledScriptPluginIntegrationTest.generated code follows kotlin-dsl coding conventions` still fails because `ktlint-convention` version `0.3.0` used by the test uses `ktlint-gradle plugin` version `7.1.0` which in turn still uses the deprecated `ProjectLayout` APIs. We need to bump it t to `8.x.x` as that version doesn't use the deprecated APIs (see commit https://github.com/JLLeitschuh/ktlint-gradle/commit/95368f667ecffa6e50513eaaa090e1b3990fd0ba).

**Comment by lptr** [comment]
https://github.com/gradle/gradle/pull/10378 is now merged, `ktlint-gradle` is upgraded. Thanks @eskatos.

File: subprojects/file-collections/src/test/groovy/org/gradle/api/internal/file/DefaultFilePropertyFactoryTest.groovy [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+
+import org.gradle.api.Task
+import org.gradle.api.internal.provider.ProviderInternal
+import org.gradle.api.internal.tasks.TaskDependencyResolveContext
+import org.gradle.test.fixtures.file.TestFile
+import org.gradle.test.fixtures.file.TestNameTestDirectoryProvider
+import org.junit.Rule
+import spock.lang.Specification
+
+class DefaultFilePropertyFactoryTest extends Specification {
```
Comment (wolfs) [inline subprojects/file-collections/src/test/groovy/org/gradle/api/internal/file/DefaultFilePropertyFactoryTest.groovy]: I wonder if these tests are already part of `FileSystemPropertySpec`.

File: subprojects/distributions/src/changes/accepted-public-api-changes.json [hunk 2]
```diff
@@ -91,6 +91,534 @@
...
+                "org.gradle.api.DefaultTask.newOutputDirectory()",
+                "org.gradle.api.DefaultTask.newOutputFile()"
+            ]
+        },
+        {
+            "type": "org.gradle.api.tasks.Delete",
+            "member": "Class org.gradle.api.tasks.Delete",
+            "acceptation": "Methods removed",
+            "changes": [
+                "org.gradle.api.DefaultTask.newInputDirectory()",
```
Comment (wolfs) [inline subprojects/distributions/src/changes/accepted-public-api-changes.json]: I should be enough to declare the removal on `DefaultTask`. Doesn't this work?

File: subprojects/file-collections/src/test/groovy/org/gradle/api/internal/file/DefaultFilePropertyFactoryTest.groovy (hunk 1 above)
Comment (lptr) [reply subprojects/file-collections/src/test/groovy/org/gradle/api/internal/file/DefaultFilePropertyFactoryTest.groovy]: There is some overlap but there are tests in both classes that are unique. This PR only moves one set around a bit. What do you suggest to do here?

File: subprojects/distributions/src/changes/accepted-public-api-changes.json (hunk 2 above)
Comment (lptr) [reply subprojects/distributions/src/changes/accepted-public-api-changes.json]: I tried that, I got this:

```text
> The following regressions are declared as accepted, but didn't match any rule:

 org.gradle.api.DefaultTask - Class org.gradle.api.DefaultTask - [org.gradle.api.DefaultTask.newInputDirectory(), org.gradle.api.DefaultTask.newInputFile(), org.gradle.api.DefaultTask.newOutputDirectory(), org.gradle.api.DefaultTask.newOutputFile()]
```

File: subprojects/file-collections/src/test/groovy/org/gradle/api/internal/file/DefaultFilePropertyFactoryTest.groovy (hunk 1 above)
Comment (wolfs) [reply subprojects/file-collections/src/test/groovy/org/gradle/api/internal/file/DefaultFilePropertyFactoryTest.groovy]: I think it is fine as is. I was wondering to merge the two test classes.

**Pull request merged**